File: src/css/properties/contain.zig
pub const css = @import("../css_parser.zig");

const SmallList = css.SmallList;

const ContainerName = css.css_rules.container.ContainerName;

const ContainerIdent = ContainerName;

/// A value for the [container-type](https://drafts.csswg.org/css-contain-3/#container-type) property.
/// Establishes the element as a query container for the purpose of container queries.
pub const ContainerType = css.DefineEnumProperty(@compileError(css.todo_stuff.depth));

/// A value for the [container-name](https://drafts.csswg.org/css-contain-3/#container-name) property.
pub const ContainerNameList = union(enum) {
    /// The `none` keyword.
    none,
    /// A list of container names.
    names: SmallList(ContainerIdent, 1),
};

/// A value for the [container](https://drafts.csswg.org/css-contain-3/#container-shorthand) shorthand property.
pub const Container = css.DefineEnumProperty(@compileError(css.todo_stuff.depth));

File: src/css/properties/contain.rs
use crate::SmallList;
use crate::css_rules::container::ContainerName;

type ContainerIdent = ContainerName;

/// A value for the [container-type](https://drafts.csswg.org/css-contain-3/#container-type) property.
/// Establishes the element as a query container for the purpose of container queries.
// TODO(port): css.DefineEnumProperty(@compileError(css.todo_stuff.depth)) — unimplemented placeholder in Zig source
pub struct ContainerType;

/// A value for the [container-name](https://drafts.csswg.org/css-contain-3/#container-name) property.
pub enum ContainerNameList {
    /// The `none` keyword.
    None,
    /// A list of container names.
    Names(SmallList<ContainerIdent, 1>),
}

/// A value for the [container](https://drafts.csswg.org/css-contain-3/#container-shorthand) shorthand property.
// TODO(port): css.DefineEnumProperty(@compileError(css.todo_stuff.depth)) — unimplemented placeholder in Zig source
pub struct Container;

// ported from: src/css/properties/contain.zig
